### PR TITLE
Audio changes

### PR DESCRIPTION
**Pull request by sticks-stuff**
Using onEnd results in a roughly 0.3 second delay from when the audio actually ends and when the event fires. I solved this for the mp3 files by simply using setTimeout for the duration of the audio clips, however the text to speech is a little more challenging to fire at the right time without using onEnd. Using onboundary fires immediately after each syllable is said so I used a npm library to calculate the syllables in the name and then compare it to how many syllables have been said so far. This results in the event being fired with no delay. If there's a discrepancy number of boundaries vs the number of syllables calculated than it will fallback onEnd anyway. 

Additionally I added an instrumental to play behind the text to speech so that there's no awkward silence and changed the voice to be male so it sounds more like the original "Michael" clip.

### CODE REVIEW

**Comment by sticks-stuff** [comment]
On an unrelated note, if you choose to accept this PR it would be cool if you could label it "hacktoberfest-accepted" <3

File: scripts.js [hunk 1]
```diff
@@ -187,36 +191,61 @@
...
     if (!hasSpeechSupport) {
       return
     }
     nameToSay = new window.SpeechSynthesisUtterance(name)
+
     nameToSay.pitch = 0.8
     nameToSay.rate = 1.25
     nameToSay.lang = 'en-US'
 
+    var voices = window.speechSynthesis.getVoices()
```
Comment (NickColley) [inline scripts.js:227]: Trying to set a deeper voice is a good idea but when I try this locally I don't get a deep voice, I imagine the voices available may be different per device?

For example my phone has a deep voice whereas my computer has a higher voice.

If you could figure out a way to have this consistent would welcome a pull request for this.

**Comment by NickColley** [comment]
I really like this idea to improve the audio so it's less juttery.

I had a go over here with Howler.js which I think the end result is seamless without too much code change:

https://github.com/nickcolley/itsyourbirthdaytoday/pull/21

I think if you could figure out the voices thing that'd be a good pull request I could accept.

Could also consider trying to do that instrumental in the background while the synthetic voice talks over it with howler by creating a new sprite for the instrumental.

**Comment by sticks-stuff** [comment]
Alright, I've changed the instrumental to use howler and I've made sure that the voice used is male on different devices (albeit through a really awful hacky solution).

File: scripts.js [hunk 2]
```diff
@@ -202,36 +207,64 @@
...
+    totalSyllables = syllable(name)
     if (!hasSpeechSupport) {
       return
     }
     nameToSay = new window.SpeechSynthesisUtterance(name)
     nameToSay.pitch = 0.8
     nameToSay.rate = 1.25
     nameToSay.lang = 'en-US'
+    var voice
+    // forgive me for i have sinned
```
Comment (NickColley) [inline scripts.js:229]: >     // forgive me for i have sinned

:joy:

File: scripts.js [hunk 3]
```diff
@@ -202,36 +207,64 @@
...
+    if (search('Microsoft David Desktop - English (United States)', window.speechSynthesis.getVoices()) !== false) { // windows
+      voice = search('Microsoft David Desktop - English (United States)', window.speechSynthesis.getVoices())
+    } else if (search('Fred (en-US)', window.speechSynthesis.getVoices()) !== false) { // ios
```
Comment (NickColley) [inline scripts.js]: You could move the `window.speechSynthesis.getVoices()` into a variable and reference it to avoid duplication.

**Comment by NickColley** [comment]
@Thestickman391 it is a little hacky but I think it's reasonable, worse case you dont find what you're searching for and it defaults to something else.

If you'd like to get this merged can you please check out the latest master version of this project and include only the voice improvements in a single pull request, I can review that on it's own as I've already merged the howler improvements. Thanks again for suggesting to improve the timings.

**Comment by sticks-stuff** [comment]
This PR should be checked out from the latest master commit though? Just looking through the diff this should only be the voice improvements  (and me slightly adjusting the timings of some sounds)

**Comment by NickColley** [comment]
I'm going to close this out as it contains a lot of unrelated unneeded changes but if you want to raise the voice improvements separately I'll consider that on its own.

**Comment by sticks-stuff** [comment]
What are the unrelated unneeded changes? I can see changing the TTS voice being unrelated and unneeded but everything else should just be fixing the awkward stuttering from the TTS voice and adding an instrumental behind it.

**Comment by NickColley** [comment]
I have merged the Howler integration that solves the awkward stuttering so it's no longer necessary.

**Comment by sticks-stuff** [comment]
There's still a delay from when the TTS voice finishes which this PR adresses

**Pull request closed without merging**